### ISSUE TITLE
fireWeapon doesn't trigger turret firing animations

### ISSUE THREAD

**Issue by DahBlount** [scp-fs2open/fs2open.github.com#1102]
Essentially, what's happening is that fireWeapon can only trigger turret fired animations, but can't properly trigger animations that occur as a turret prepares to fire, like doors on missile bays opening. My current best guess as to a possible fix is to expose functionality to check subsystem animation positions from LUA so that scripters can call fireWeapon only once the animation has been fully run.

Other than that, I'm open to suggestions.

**Comment by asarium** [comment]
There is the `getAnimationDoneTime` function in the `ship` type but that is probably not enough. The major problem with the animation code is that no one understands how it works so adding features on top of that is unlikely to happen until it is refactored.

**Comment by DahBlount** [comment]
Well, I suppose it's a good thing I've actually got some notes on the queued_animation system. I'll need to do a bit more exploring, but I've looked at the big parts that need replacing or adjustment to allow for animations other than rotations (ie. translations, scalars, pre-baked, skeleton).

**Comment by chief1983** [comment]
I was always under the impression that this was by design, and it was up to the person using fireWeapon to run the animation sequences before firing the weapon.  I also think this is used this way in released content.  I didn't even think this could rotate a turret to be pointing in the right direction.

**Comment by asarium** [comment]
Yes, `fireWeapon` was never intended to be a fully self-contained solution for firing a turret. However, currently there is no way for Lua to actually trigger the animation sequences that may need to be done before a turret is fired.

**Comment by DahBlount** [comment]
I was wondering if we could expose the animation code to 'l_Subsytem' as there is a function specifically for subsystems that plays animations. We could allow scripters to check the animation position enums as well to determine when an animation has finished as well as auto-play the reverse of the animation.

**Comment by EatThePath** [comment]
@TRBlount @BMagnu do the recent changes to animation provide the functionality discussed here?

**Comment by MjnMixael** [comment]
I'm closing this issue as the newer animation system does provide the missing scripting capability for triggering turret firing animations.
